src/main/java/com/ood/myorange/controllor/DownloadController.java

```diff
@@ -0,0 +1,38 @@
+package com.ood.myorange.controllor;
+
+import com.ood.myorange.dao.StorageConfigDao;
+import com.ood.myorange.dto.response.BaseResponse;
+import com.ood.myorange.pojo.StorageConfig;
+import com.ood.myorange.util.ModelMapperUtil;
+import com.ood.myorange.util.StorageUtilFactory;
+import com.ood.myorange.util.storageUtils.StorageUtil;
+import com.oracle.tools.packager.Log;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import static com.ood.myorange.dto.response.BaseResponse.success;
+
+@RestController
+@Slf4j
+public class DownloadController {
+
+    @Autowired
+    StorageUtilFactory storageUtilFactory;
+
+    @Autowired
+    StorageConfigDao storageConfigDao;
+
+    // This just an example
+    // Not working
+    @GetMapping("/d/{key}")
+    public BaseResponse getPreSignedURL(@PathVariable("key") int key) {
```

Comment:
What is the key?

Reply:
it should be the file_id in the `user_file` table.

Reply:
I think it is not a good idea to use the file_id in our database, you can think about how the sdk get download url for a file, does it using file name? or other params?  

In this way, the programmer should fetch the real file name from the DB and pass it to your function.

Reply:
Yep, you are correct.
So the user will pass the file name to the controller, and I will obtain the user_id in the authenticationfacade.
So that I can get the file_id in the db using the mentioned params.

Reply:
you don't need the user to pass their name, the authenticationfacade can get the current login user's info, its like an identity context. It is varied in different threads.